### DISCUSSION TITLE
TypeError: Cannot read properties of undefined (reading '_internal')

### DISCUSSION THREAD

**Discussion by maccman** [xaviergonz/mobx-keystone#480] [Q&A]
@xaviergonz I'm curious if you might know what this is off the top of your head? We're seeing it in prod too.

<img width="969" alt="CleanShot 2022-08-08 at 12 26 00@2x" src="https://user-images.githubusercontent.com/2142/183466937-5a5b7cb2-60e4-4764-926a-d684e8f23b36.png">

**Comment by xaviergonz** [comment]
do you at least have a longer stack trace?

**Reply by maccman** [reply]
<img width="957" alt="CleanShot 2022-08-08 at 12 52 11@2x" src="https://user-images.githubusercontent.com/2142/183471332-af25c29b-eb32-4a71-8b44-10b4d97b9c10.png">

**Reply by xaviergonz** [reply]
are you sure that stack trace is for the same error?
it shouldn't be able to reach line 519 if propData were to be undefined, since it should have failed in line 518 already...

**Reply by maccman** [reply]
I'm fairly sure this it the same error - but unfortunately I'm struggling to reproduce it in dev.

As you say, very peculiar. If _internal was undefined then we'd error on 518. It's possible Sentry is getting the lines one off.

**Reply by xaviergonz** [reply]
My hunch is that it might be related with applying snapshots that have extra properties which are not declared in the model, but I'm checking.

**Reply by xaviergonz** [reply]
got it, should be fixed now in v0.69.9

**Reply by xaviergonz** [reply]
If you are curious, the unit test in this commit was triggering it
https://github.com/xaviergonz/mobx-keystone/commit/0a6196529b077e48d1d5580762ee77e5791d9713

**Reply by maccman** [reply]
I do believe you are the fastest maintainer I have ever seen - holy smokes. Thank you!
